docs/reference/creating_a_provider_class.rst

```diff
@@ -253,29 +253,30 @@ Register the Class with the Service Container
 If you use the tag ``sonata.media.provider``, the provider service will be
 added to the provider pool.
 
-.. code-block:: xml
-
-    <!-- config/services.xml -->
-
-    <service id="sonata.media.provider.vimeo" class="Sonata\MediaBundle\Provider\VimeoProvider">
-        <argument>sonata.media.provider.vimeo</argument>
-        <argument type="service" id="sonata.media.filesystem.local"/>
-        <argument type="service" id="sonata.media.cdn.server"/>
-        <argument type="service" id="sonata.media.generator.default"/>
-        <argument type="service" id="sonata.media.thumbnail.format"/>
-        <argument type="service" id="sonata.media.buzz.browser"/>
-        <argument type="service" id="sonata.media.metadata.proxy"/>
-        <call method="setTemplates">
-            <argument type="collection">
-                <argument key='helper_thumbnail'>@SonataMedia/Provider/thumbnail.html.twig</argument>
-                <argument key='helper_view'>@SonataMedia/Provider/view_vimeo.html.twig</argument>
-            </argument>
-        </call>
-        <call method="setResizer">
-            <argument type="service" id="sonata.media.resizer.simple"/>
-        </call>
-        <tag name="sonata.media.provider"/>
-    </service>
+.. code-block:: yaml
+
+    # config/services.yaml
+
+    sonata.media.provider.vimeo:
+        class: Sonata\MediaBundle\Provider\VimeoProvider
+
+        arguments:
+            - sonata.media.provider.vimeo
+            - '@sonata.media.filesystem.local'
+            - '@sonata.media.cdn.server'
+            - '@sonata.media.generator.default'
+            - '@sonata.media.thumbnail.format'
+            - '@sonata.media.buzz.browser'
+            - '@sonata.media.metadata.proxy'
+
+        calls:
+            - [setTemplates, [{ helper_thumbnail: '@@SonataMedia/Provider/thumbnail.html.twig', helper_view: '@@SonataMedia/Provider/view_vimeo.html.twig' }]]
```

Comment:
Can you please use a multiline equivalent here? Overall YAML is easier to read here, but this isn't, especially with the double `@`…

Reply:
Hi,

Yaml is not really well suited for this kind of structure, this is the best I could come up with:

``` yaml
        calls:
        -
            - setTemplates
            -
                -
                    helper_thumbnail: @@SonataMedia/Provider/thumbnail.html.twig'
                    helper_view: '@@SonataMedia/Provider/view_vimeo.html.twig'
        -
            - setResizer
            -
                - '@sonata.media.resizer.simple'
```

or more concise but even less understandable

```yaml
        calls:
        -
            - setTemplates
            - - helper_thumbnail: '@@SonataMedia/Provider/thumbnail.html.twig'
                helper_view: '@@SonataMedia/Provider/view_vimeo.html.twig'
```

I think this looks a bit crazy. :)

Either we leave as is, or I just update the xml version.
But Symfony removed most docs for the xml configs, so it would be good to migrate to yaml anyway...

Reply:
> I think this looks a bit crazy. :)

Maybe that's because you are not used to multiline syntax? The second one looks sensible to me. More than what you originally proposed anyway.

Reply:
I updated the docs with the multi line syntax.